app/server/identity/identity.ts

```diff
@@ -31,52 +32,94 @@ export const getUser: (
   }
 
   const cookieJar = parseCookie(cookies);
-  const GU_U = cookieJar[keys.GU];
-  const SC_GU_U = cookieJar[keys.SC];
+  const GU_U = cookieJar[keys.GU_U];
+  const SC_GU_U = cookieJar[keys.SC_GU_U];
+  const SC_GU_LA = cookieJar[keys.SC_GU_LA];
 
-  if (GU_U == null || SC_GU_U == null) {
+  if (GU_U == null || SC_GU_U == null || SC_GU_LA == null) {
     return IdentityError.NotLoggedIn;
   }
 
-  const [encodedToken] = SC_GU_U.split(".");
-  const cookieString = safely(() => decode(encodedToken));
-  if (cookieString == null) {
-    return IdentityError.CouldNotParse;
+  const scGuUExpiryError = checkScGuUExpiry(SC_GU_U);
+  if (scGuUExpiryError !== undefined) {
+    return scGuUExpiryError;
+  }
+
+  const scGuLaExpiryError = checkScGuLaExpiry(SC_GU_LA);
+  if (scGuLaExpiryError !== undefined) {
+    return scGuLaExpiryError;
   }
-  const parsed = safely(() => JSON.parse(cookieString));
-  if (parsed == null) {
+
+  return {
+    GU_U,
+    SC_GU_U,
+    SC_GU_LA
+  };
+};
+
+const safely: <T>(f: () => T) => T | null = f => {
+  try {
+    return f();
+  } catch (e) {
+    return null;
+  }
+};
+
+export const checkScGuUExpiry: (
+  SC_GU_U: string
+) => IdentityError | undefined = (SC_GU_U: string) => {
+  const scGuGuStr = safely(() => decode(SC_GU_U.split(".", 1)[0]));
+  if (scGuGuStr == null) {
     return IdentityError.CouldNotParse;
   }
-  if (!(Array.isArray(parsed) && parsed.length === 2)) {
+  const scGuGuParsed = safely(() => JSON.parse(scGuGuStr));
+  if (
+    scGuGuParsed == null ||
+    !Array.isArray(scGuGuParsed) ||
+    scGuGuParsed.length !== 2
+  ) {
     return IdentityError.CouldNotParse;
   }
-
-  const [id, expires] = parsed;
+  const [identityID, expires] = scGuGuParsed;
   if (expires == null) {
     return IdentityError.CouldNotParse;
   }
-
   const expiry = safely(() => parseInt(expires, 10));
   if (expiry == null) {
     return IdentityError.CouldNotParse;
   }
-
   const remaining = expiry - new Date().getTime();
   if (remaining < ONE_HOUR) {
     return IdentityError.Expired;
   }
-
-  return {
-    GU_U,
-    SC_GU_U,
-    expiry
-  };
 };
 
-const safely: <T>(f: () => T) => T | null = f => {
-  try {
-    return f();
-  } catch (e) {
-    return null;
+export const checkScGuLaExpiry: (
+  SC_GU_LA: string
+) => IdentityError | undefined = (SC_GU_LA: string) => {
+  const scGuLaStr = safely(() => decode(SC_GU_LA.split(".")[0]));
+  if (scGuLaStr == null) {
+    return IdentityError.CouldNotParse;
+  }
+  const scGuLaParsed = safely(() => JSON.parse(scGuLaStr));
+  if (
+    scGuLaParsed == null ||
+    !Array.isArray(scGuLaParsed) ||
+    scGuLaParsed.length !== 3
+  ) {
+    return IdentityError.CouldNotParse;
```

Comment:
If we fail to parse, do we ask users to re-authenticate?

Reply:
if we fail to parse - it is considered as 'not logged in' which redirects to signin rather than reauthenticate (see https://github.com/guardian/manage-frontend/blob/45b74574c41ef3cd2cbbe9d4d77c656d1225ba59/app/server/identity/identityMiddleware.ts#L26)